src/App.js

```diff
@@ -17,6 +18,7 @@ function App() {
             <Route path="/" element={<Navigate to="/articles" />} />
             <Route path="/articles/:topic" element={<Main />} />
             <Route path="/articles" element={<Main />} />
+            <Route path="/articles/:topic/:article_id" element={<SingleArticle/>}/>
```

Comment:
this isn't a huge problem but be aware that `/articles/not-a-topic/1` will successfully render article 1!

Thankfully your back button still sends to the correct topic page